File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    ponkemons = {}
    
    for n in nums:
        if n in ponkemons.keys():
            ponkemons[n] += 1
            
        else:
            ponkemons[n] = 1
            
    if len(list(ponkemons.keys())) <= len(nums) // 2:
        answer = len(list(ponkemons.keys()))
        
    else:
        answer = len(nums) // 2
        
    return answer
    

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <algorithm>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums)
{
    int answer = 0;
    unordered_set<int> us(nums.begin(), nums.end());
    
    return min(us.size(), nums.size() / 2);
}